soccer/gameplay/skills/pivot_kick.py

```diff
@@ -123,6 +131,48 @@ def facing_opp_goal(self):
 
         return False
 
+    def opp_robot_blocking(self):
+        if (self.robot is None):
+            return False
+
+        # Closest opp robot in any direction
+        # To us, not the ball   
+        closest_opp_robot = None
+        closest_opp_dist = float("inf")
+        for r in main.their_robots():
+            if ((r.pos - self.robot.pos).mag() < closest_opp_dist):
+                closest_opp_robot = r
+                closest_opp_dist = (r.pos - self.robot.pos).mag()
+
+        # Only do this if a robot is in range
+        robot_in_range = closest_opp_dist < .2 + 2 * constants.Robot.Radius
+
+        aim_dir = robocup.Point.direction(self.robot.angle)
+        robot_dir = (closest_opp_robot.pos - self.robot.pos)
+
+        # Only trigger if they are infront of us
+        robot_in_front = aim_dir.dot(robot_dir) > 0
+
+        closest_pt = robocup.Line(
+            self.robot.pos,
+            self.robot.pos + aim_dir).nearest_point(closest_opp_robot.pos)
+
+        does_hit_robot = (closest_opp_robot.pos - closest_pt
+                          ).mag() < constants.Robot.Radius
+
+        facing_their_side = robocup.Point.direction(self.robot.angle).y > 0
+
+        ret = (facing_their_side and robot_in_range and robot_in_front and
+               does_hit_robot)
+
+        if ret:
+            print("Panic kick")
+            main.debug_drawer().draw_text('panic kick', self.robot.pos,
+                                          (255, 255, 255), 'PivotKick')
+
```

Comment:
I recall us discussing after comp adding a 'panic chip' based on opponent distance and direction of our robot.

Reply:
That would be a separate issue, right?